src/audio/copier/copier_dai.c

```diff
@@ -64,6 +64,78 @@ static int copier_set_alh_multi_gtw_channel_map(struct comp_dev *dev,
 	return 0;
 }
 
+static int copier_alh_assign_dai_index(struct comp_dev *dev,
+				       void *gtw_cfg_data,
+				       union ipc4_connector_node_id node_id,
+				       struct ipc_config_dai *dai,
+				       int *dai_index,
+				       int *dai_count)
+{
+	struct processing_module *mod = comp_get_drvdata(dev);
+	struct copier_data *cd = module_get_private_data(mod);
+	const struct sof_alh_configuration_blob *alh_blob = gtw_cfg_data;
+	int i, dai_num;
+
+	if (!cd->config.gtw_cfg.config_length) {
+		comp_err(mod->dev, "No gateway config found in blob!");
+		return -EINVAL;
+	}
+
+#if defined(CONFIG_ACE_VERSION_2_0)
```

Comment:
Here again, we have very Intel specific code in copier_dai.c which would need to be generic. We've had DAI type specific code segments before in src/audio , but this is expanding the amount of code quite significantly.